### PR TITLE
Implement for_each and use it in the example

### PR DESCRIPTION
**Pull request by tpadioleau**

### CODE REVIEW

File: include/for_each.hpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    DiscreteDomainIterator<DDim2> const it2_e = select<DDim2>(domain).end();
+#pragma omp parallel for collapse(2) default(none) shared(it1_b, it1_e, it2_b, it2_e, f)
+    for (DiscreteDomainIterator<DDim1> it1 = it1_b; it1 != it1_e; ++it1) {
+        for (DiscreteDomainIterator<DDim2> it2 = it2_b; it2 != it2_e; ++it2) {
+            f(DiscreteCoordinate<DDim1, DDim2>(*it1, *it2));
+        }
+    }
+}
+
+using default_policy = serial_policy;
```
Comment (jbigot) [inline include/for_each.hpp:85]: you could create a constexpr instance for the users to use

File: include/for_each.hpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include <utility>
+
+#include "ddc/chunk_span.hpp"
+#include "ddc/discrete_coordinate.hpp"
+#include "ddc/discrete_domain.hpp"
+
+/// Serial execution
+struct serial_policy
```
Comment (jbigot) [inline include/for_each.hpp:10]: you could create a constexpr instance for the users to use

File: include/for_each.hpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            f(DiscreteCoordinate<DDim1, DDim2>(i1, i2));
+        }
+    }
+}
+
+/** OpenMP parallel execution
+ * - default scheduling
+ * - collapsing in case of tightly nested loops
+ */
+struct omp_policy
```
Comment (jbigot) [inline include/for_each.hpp:47]: you could create a constexpr instance for the users to use

**Review by jbigot — APPROVED** [review]

File: include/for_each.hpp (hunk 2 above)
Comment (tpadioleau) [reply include/for_each.hpp:10]: Done

File: include/for_each.hpp (hunk 3 above)
Comment (tpadioleau) [reply include/for_each.hpp:47]: Done

File: include/for_each.hpp (hunk 1 above)
Comment (tpadioleau) [reply include/for_each.hpp:85]: I would not expose it for now.

File: docs/first_steps/heat_equation.cpp [hunk 4]
```diff
@@ -1,6 +1,7 @@
 #include <ddc/Chunk>
 #include <ddc/DiscreteCoordinate>
 #include <ddc/DiscreteDomain>
+#include <ddc/ForEach>
```
[the remark this answers is not in the document]
Comment (jbigot) [reply docs/first_steps/heat_equation.cpp:4]: `ddc/for_each`

**Comment by jbigot** [comment]
@tpadioleau : we're good, aren't we?

**Comment by tpadioleau** [comment]
I think so, let's merge.

**Pull request merged**